src/fsharp/vs/service.fs

```diff
@@ -1372,153 +1414,127 @@ module internal Parser =
             // Sanity check here. The phase of an error should be in a phase known to the language service.
             let exn =
                 if not(exn.IsPhaseInCompile()) then
                     // Reaching this point means that the error would be sticky if we let it prop up to the language service.
                     // Assert and recover by replacing phase with one known to the language service.
                     Trace.TraceInformation(sprintf "The subcategory '%s' seen in an error should not be seen by the language service" (exn.Subcategory()))
-                    {exn with Phase=BuildPhase.TypeCheck}
+                    { exn with Phase = BuildPhase.TypeCheck }
                 else exn
-            if reportErrors then 
-                let report exn = 
-                    for ei in ErrorHelpers.ReportError (tcConfig, false, mainInputFileName, fileInfo, (exn, sev)) do
+            if reportErrors then
+                let report exn =
+                    for ei in ErrorHelpers.ReportError (options, false, mainInputFileName, fileInfo, (exn, sev)) do
                         errorsAndWarningsCollector.Add ei
-                        if sev = FSharpErrorSeverity.Error then 
+                        if sev = FSharpErrorSeverity.Error then
                             errorCount <- errorCount + 1
-                      
+
                 match exn with
 #if EXTENSIONTYPING
-                | {Exception = (:? TypeProviderError as tpe)} ->
-                    tpe.Iter (fun e ->
-                        let newExn = {exn with Exception = e}
-                        report newExn
-                    )
+                | { Exception = (:? TypeProviderError as tpe) } -> tpe.Iter(fun e -> report { exn with Exception = e })
 #endif
                 | e -> report e
-      
-        let errorLogger = 
-            { new ErrorLogger("ErrorHandler") with 
+
+        let errorLogger =
+            { new ErrorLogger("ErrorHandler") with
                 member x.DiagnosticSink (exn, isError) = diagnosticSink (if isError then FSharpErrorSeverity.Error else FSharpErrorSeverity.Warning) exn
                 member x.ErrorCount = errorCount }
-      
-      
+
         // Public members
         member x.ErrorLogger = errorLogger
         member x.CollectedDiagnostics = errorsAndWarningsCollector.ToArray()
         member x.ErrorCount = errorCount
-        member x.TcConfig with set tc = tcConfig <- tc
+        member x.ErrorSeverityOptions with set opts = options <- opts
         member x.AnyErrors = errorCount > 0
 
+    let getLightSyntaxStatus fileName options =
+        let lower = String.lowercase fileName
+        let lightOnByDefault = List.exists (Filename.checkSuffix lower) FSharpLightSyntaxFileSuffixes
+        let lightSyntaxStatus = if lightOnByDefault then (options.LightSyntax <> Some false) else (options.LightSyntax = Some true)
+        LightSyntaxStatus(lightSyntaxStatus, true)
```

Comment:
It feels like this code is duplicating ``ComputeLightSyntaxInitialStatus`` ?
 https://github.com/Microsoft/visualfsharp/blob/20573b7decc4989b625a769952c45a4c53fc510a/src/fsharp/CompileOps.fs#L3010